### PR TITLE
Fix Trying to access array offset on value of type null

### LINKED ISSUE #32: Trying to access array offset on value of type null in Doctrine\Common\Reflection\StaticReflectionParser->parse
I'm getting the following notice running version 1.1.0:
```
Notice: Trying to access array offset on value of type null in Doctrine\Common\Reflection\StaticReflectionParser->parse() (line 192).
```

First part of the stack trace:
```
Doctrine\Common\Reflection\StaticReflectionParser->parse() (Line: 303)
Doctrine\Common\Reflection\StaticReflectionParser->getDocComment() (Line: 35)
Doctrine\Common\Reflection\StaticReflectionClass->getDocComment() (Line: 101)
Drupal\Component\Annotation\Doctrine\SimpleAnnotationReader->getClassAnnotations(Object) (Line: 125)
```

### PR DESCRIPTION
**Pull request by DieterHolvoet**
Fixes #32

### CODE REVIEW

**Review by SenseException — CHANGES_REQUESTED** [review]
Please add test(s) that are covering your `null` case.

**Comment by DieterHolvoet** [comment]
It's my first time writing a test, so let me know if anything is off.

File: tests/Doctrine/Tests_PHP74/Common/Reflection/Dummies/TokenParserNullValueTestClass.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+
+namespace Doctrine\Tests_PHP74\Common\Reflection\Dummies;
+
+trait TokenParserNullValueTestClass
```
Comment (malarzm) [inline tests/Doctrine/Tests_PHP74/Common/Reflection/Dummies/TokenParserNullValueTestClass.php]: Am I correct assuming this was previously failing due to `static function ($value)` having no name specified? If so then please rename the trait and test to make it obvious what's "wrong"

Comment (DieterHolvoet) [reply tests/Doctrine/Tests_PHP74/Common/Reflection/Dummies/TokenParserNullValueTestClass.php]: I changed the name, do you think it's better like this?

Comment (malarzm) [reply tests/Doctrine/Tests_PHP74/Common/Reflection/Dummies/TokenParserNullValueTestClass.php]: Way better :+1:

**Comment by DieterHolvoet** [comment]
Anything left to do here?

File: tests/Doctrine/Tests_PHP74/Common/Reflection/TokenParserAnonymousFunctionTest.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public function testGetValue() : void
+    {
+        $testsRoot              = substr(__DIR__, 0, -strlen(__NAMESPACE__) - 1);
+        $paths                  = [
+            'Doctrine\\Tests_PHP74' => [$testsRoot],
+        ];
+        $staticReflectionParser = new StaticReflectionParser(TokenParserAnonymousFunctionTestClass::class, new Psr0FindFile($paths));
+
+        $this->assertEquals('', $staticReflectionParser->getDocComment());
```
Comment (SenseException) [inline tests/Doctrine/Tests_PHP74/Common/Reflection/TokenParserAnonymousFunctionTest.php]: A small detail:
```suggestion
        self::assertEquals('', $staticReflectionParser->getDocComment());
```

**Review by SenseException — CHANGES_REQUESTED** [review]
Looks good so far.

I'd like to have another Doctrine reviewer for this PR as a second pair of eyes.

**Comment by greg0ire** [comment]
Please kindly squash your commits together. If you don't, we'll try to remember to do it for you but it's best if you save us this trouble.

How to do that?

1. `git rebase -i origin/3.x`, assuming `origin` is a git remote that points to this repository, and not your fork. If you're not sure what your remotes are, run `git remote -vvv`, there should be your fork and the holy/reference/base/origin/whatever-you-call-it repository.
2. A window will show up with many lines, replace `pick` with `fixup` on every line but the first one
3. Close your editor, git should do its magic, and you should end up with one commit
4. Use `git push --force` to overwrite what you already push. Don't forget the `--force` option otherwise git will try to merge both things together.

**Comment by DieterHolvoet** [comment]
@greg0ire I should have squashed my commits, let me know if I did anything wrong.

**Comment by greg0ire** [comment]
It's better 👍 On to the next issue: the commit message itself.

Please improve your commit message according to [the contributing guide](https://www.doctrine-project.org/contribute/index.html#working-on-topic-branches).

**Comment by DieterHolvoet** [comment]
Changed the commit message.

**Review by greg0ire — APPROVED** [review]

**Review by SenseException — APPROVED** [review]

**Comment by DieterHolvoet** [comment]
Any update on this?

**Comment by malarzm** [comment]
@DieterHolvoet sorry for long time wait! I've changed the base branch to `1.1.x` since that's the branch you've found the error on and it's still maintained, but now PR has additional commits. Could you please rebase your work onto `1.1.x`? Please ping me once that's done and I'll merge it right away :)

**Comment by DieterHolvoet** [comment]
@malarzm Should be done now

**Comment by greg0ire** [comment]
There are commits that do not belong to you in here. I'm going to rebase your branch.

**Comment by greg0ire** [comment]
Ah you didn't allow edit from maintainers, so I cannot. Can you please use `git reset --hard origin/1.1.x && git cherry-pick d2ca550 && git push -f`?

**Comment by DieterHolvoet** [comment]
I did allow edits from maintainers, but I'll try. I wonder what I did wrong...

**Comment by greg0ire** [comment]
Thanks @DieterHolvoet !
